featurebyte/query_graph/node/metadata/sdk_code.py

```diff
@@ -421,6 +426,40 @@ def convert_to_variable_name(
         return var_name
 
 
+class UnusedVariableFinder(ast.NodeVisitor):
+    """UnusedVariableFinder class is used to find the unused variables in the generated SDK code."""
+
+    def __init__(self) -> None:
+        self.variables: Set[str] = set()
+        self.used_variables: Set[str] = set()
+
+    def visit_Name(self, node: ast.Name) -> None:  # pylint: disable=invalid-name
+        """
+        Visit the Name node in the ast.
+
+        Parameters
+        ----------
+        node: ast.Name
+            Ast name node
+        """
+        if isinstance(node.ctx, ast.Store):
+            # add variable to set of variables
+            self.variables.add(node.id)
+        elif isinstance(node.ctx, ast.Load):
+            # add variable to set of used variables
+            self.used_variables.add(node.id)
```

Comment:
Curious how does this compare with using something more tested like `vulture`? Will there be any special cases that this doesn't handle?

Reply:
My understanding is that `vulture` is using something similar here: https://github.com/jendrikseipp/vulture/blob/main/vulture/core.py#L645-L652

Reply:
Here we did not consider `ast.Del` and `ast.Param` (would expect `ast.Del` handle statement like `del var` and `ast.Param` is to handle the function or method's parameter name).

Reply:
Will update this part to make it similar to `vulture` and add test to cover the other 2 cases.